// ng2-table/__ivy_ngcc__/components/table/ng-table-filtering.directive.js
"use strict";
var core_1 = require('@angular/core');
// import {setProperty} from 'angular2/ts/src/core/forms/directives/shared';
function setProperty(renderer, elementRef, propName, propValue) {
    renderer.setElementProperty(elementRef, propName, propValue);
}
var ɵngcc0 = require('@angular/core');
var NgTableFilteringDirective = (function () {
    function NgTableFilteringDirective(element, renderer) {
        this.ngTableFiltering = {
            filterString: '',
            columnName: 'name'
        };
        this.tableChanged = new core_1.EventEmitter();
        this.element = element;
        this.renderer = renderer;
        // Set default value for filter
        setProperty(this.renderer, this.element, 'value', this.ngTableFiltering.filterString);
    }
    Object.defineProperty(NgTableFilteringDirective.prototype, "config", {
        get: function () {
            return this.ngTableFiltering;
        },
        set: function (value) {
            this.ngTableFiltering = value;
        },
        enumerable: true,
        configurable: true
    });
    NgTableFilteringDirective.prototype.onChangeFilter = function (event) {
        this.ngTableFiltering.filterString = event;
        this.tableChanged.emit({ filtering: this.ngTableFiltering });
    };
    /** @nocollapse */
    NgTableFilteringDirective.ctorParameters = [
        { type: core_1.ElementRef, },
        { type: core_1.Renderer, },
    ];
    NgTableFilteringDirective.propDecorators = {
        'ngTableFiltering': [{ type: core_1.Input },],
        'tableChanged': [{ type: core_1.Output },],
        'config': [{ type: core_1.Input },],
        'onChangeFilter': [{ type: core_1.HostListener, args: ['input', ['$event.target.value'],] },],
    };
NgTableFilteringDirective.ɵfac = function NgTableFilteringDirective_Factory(t) { return new (t || NgTableFilteringDirective)(ɵngcc0.ɵɵdirectiveInject(core_1.ElementRef), ɵngcc0.ɵɵdirectiveInject(core_1.Renderer)); };
NgTableFilteringDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NgTableFilteringDirective, selectors: [["", "ngTableFiltering", ""]], hostBindings: function NgTableFilteringDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("input", function NgTableFilteringDirective_input_HostBindingHandler($event) { return ctx.onChangeFilter($event.target.value); });
    } }, inputs: { ngTableFiltering: "ngTableFiltering", config: "config" }, outputs: { tableChanged: "tableChanged" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTableFilteringDirective, [{
        type: core_1.Directive,
        args: [{ selector: '[ngTableFiltering]' }]
    }], function () { return [{ type: core_1.ElementRef }, { type: core_1.Renderer }]; }, { ngTableFiltering: [{
            type: core_1.Input
        }], tableChanged: [{
            type: core_1.Output
        }], config: [{
            type: core_1.Input
        }], onChangeFilter: [{
            type: core_1.HostListener,
            args: ['input', ['$event.target.value']]
        }] }); })();
    return NgTableFilteringDirective;
}());
exports.NgTableFilteringDirective = NgTableFilteringDirective;
