// ng2-table/__ivy_ngcc__/components/table/ng-table-paging.directive.js
"use strict";
var core_1 = require('@angular/core');
var ɵngcc0 = require('@angular/core');
var NgTablePagingDirective = (function () {
    function NgTablePagingDirective() {
        this.ngTablePaging = true;
        this.tableChanged = new core_1.EventEmitter();
    }
    Object.defineProperty(NgTablePagingDirective.prototype, "config", {
        get: function () {
            return this.ngTablePaging;
        },
        set: function (value) {
            this.ngTablePaging = value;
        },
        enumerable: true,
        configurable: true
    });
    NgTablePagingDirective.prototype.onChangePage = function (event) {
        // Object.assign(this.config, event);
        if (this.ngTablePaging) {
            this.tableChanged.emit({ paging: event });
        }
    };
    /** @nocollapse */
    NgTablePagingDirective.ctorParameters = [];
    NgTablePagingDirective.propDecorators = {
        'ngTablePaging': [{ type: core_1.Input },],
        'tableChanged': [{ type: core_1.Output },],
        'config': [{ type: core_1.Input },],
        'onChangePage': [{ type: core_1.HostListener, args: ['pagechanged', ['$event'],] },],
    };
NgTablePagingDirective.ɵfac = function NgTablePagingDirective_Factory(t) { return new (t || NgTablePagingDirective)(); };
NgTablePagingDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NgTablePagingDirective, selectors: [["", "ngTablePaging", ""]], hostBindings: function NgTablePagingDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("pagechanged", function NgTablePagingDirective_pagechanged_HostBindingHandler($event) { return ctx.onChangePage($event); });
    } }, inputs: { ngTablePaging: "ngTablePaging", config: "config" }, outputs: { tableChanged: "tableChanged" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTablePagingDirective, [{
        type: core_1.Directive,
        args: [{ selector: '[ngTablePaging]' }]
    }], function () { return []; }, { ngTablePaging: [{
            type: core_1.Input
        }], tableChanged: [{
            type: core_1.Output
        }], config: [{
            type: core_1.Input
        }], onChangePage: [{
            type: core_1.HostListener,
            args: ['pagechanged', ['$event']]
        }] }); })();
    return NgTablePagingDirective;
}());
exports.NgTablePagingDirective = NgTablePagingDirective;
