// ng2-table/__ivy_ngcc__/components/table/ng-table-sorting.directive.js
"use strict";
var core_1 = require('@angular/core');
var ɵngcc0 = require('@angular/core');
var NgTableSortingDirective = (function () {
    function NgTableSortingDirective() {
        this.sortChanged = new core_1.EventEmitter();
    }
    Object.defineProperty(NgTableSortingDirective.prototype, "config", {
        get: function () {
            return this.ngTableSorting;
        },
        set: function (value) {
            this.ngTableSorting = value;
        },
        enumerable: true,
        configurable: true
    });
    NgTableSortingDirective.prototype.onToggleSort = function (event) {
        if (event) {
            event.preventDefault();
        }
        if (this.ngTableSorting && this.column && this.column.sort !== false) {
            switch (this.column.sort) {
                case 'asc':
                    this.column.sort = 'desc';
                    break;
                case 'desc':
                    this.column.sort = '';
                    break;
                default:
                    this.column.sort = 'asc';
                    break;
            }
            this.sortChanged.emit(this.column);
        }
    };
    /** @nocollapse */
    NgTableSortingDirective.ctorParameters = [];
    NgTableSortingDirective.propDecorators = {
        'ngTableSorting': [{ type: core_1.Input },],
        'column': [{ type: core_1.Input },],
        'sortChanged': [{ type: core_1.Output },],
        'config': [{ type: core_1.Input },],
        'onToggleSort': [{ type: core_1.HostListener, args: ['click', ['$event'],] },],
    };
NgTableSortingDirective.ɵfac = function NgTableSortingDirective_Factory(t) { return new (t || NgTableSortingDirective)(); };
NgTableSortingDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NgTableSortingDirective, selectors: [["", "ngTableSorting", ""]], hostBindings: function NgTableSortingDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function NgTableSortingDirective_click_HostBindingHandler($event) { return ctx.onToggleSort($event); });
    } }, inputs: { config: "config", ngTableSorting: "ngTableSorting", column: "column" }, outputs: { sortChanged: "sortChanged" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTableSortingDirective, [{
        type: core_1.Directive,
        args: [{ selector: '[ngTableSorting]' }]
    }], function () { return []; }, { sortChanged: [{
            type: core_1.Output
        }], config: [{
            type: core_1.Input
        }], onToggleSort: [{
            type: core_1.HostListener,
            args: ['click', ['$event']]
        }], ngTableSorting: [{
            type: core_1.Input
        }], column: [{
            type: core_1.Input
        }] }); })();
    return NgTableSortingDirective;
}());
exports.NgTableSortingDirective = NgTableSortingDirective;
