// ng2-table/__ivy_ngcc__/components/table/ng-table.component.js
"use strict";
var core_1 = require('@angular/core');
var platform_browser_1 = require('@angular/platform-browser');
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/common');
var ɵngcc2 = require('./ng-table-sorting.directive');
var ɵngcc3 = require('./ng-table-filtering.directive');

var _c0 = function (a0, a1) { return { "fa-chevron-down": a0, "fa-chevron-up": a1 }; };
function NgTableComponent_th_3_i_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "i", 7);
} if (rf & 2) {
    var column_r3 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(1, _c0, column_r3.sort === "desc", column_r3.sort === "asc"));
} }
function NgTableComponent_th_3_Template(rf, ctx) { if (rf & 1) {
    var _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "th", 5);
    ɵngcc0.ɵɵlistener("sortChanged", function NgTableComponent_th_3_Template_th_sortChanged_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r7); var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.onChangeTable($event); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, NgTableComponent_th_3_i_2_Template, 1, 4, "i", 6);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r3 = ctx.$implicit;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("ngClass", column_r3.className || "");
    ɵngcc0.ɵɵproperty("ngTableSorting", ctx_r0.config)("column", column_r3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", column_r3.title, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.config && column_r3.sort);
} }
function NgTableComponent_tr_5_td_1_input_1_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 9);
    ɵngcc0.ɵɵlistener("tableChanged", function NgTableComponent_tr_5_td_1_input_1_Template_input_tableChanged_0_listener() { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r11 = ɵngcc0.ɵɵnextContext(3); return ctx_r11.onChangeTable(ctx_r11.config); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r9 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", column_r9.filtering.placeholder);
    ɵngcc0.ɵɵproperty("ngTableFiltering", column_r9.filtering);
} }
function NgTableComponent_tr_5_td_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td");
    ɵngcc0.ɵɵtemplate(1, NgTableComponent_tr_5_td_1_input_1_Template, 1, 2, "input", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r9 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", column_r9.filtering);
} }
function NgTableComponent_tr_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵtemplate(1, NgTableComponent_tr_5_td_1_Template, 2, 1, "td", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.columns);
} }
function NgTableComponent_tr_6_td_1_Template(rf, ctx) { if (rf & 1) {
    var _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 11);
    ɵngcc0.ɵɵlistener("click", function NgTableComponent_tr_6_td_1_Template_td_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r19); var column_r16 = ctx.$implicit; var row_r14 = ɵngcc0.ɵɵnextContext().$implicit; var ctx_r17 = ɵngcc0.ɵɵnextContext(); return ctx_r17.cellClick(row_r14, column_r16.name); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r16 = ctx.$implicit;
    var row_r14 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r15 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("innerHtml", ctx_r15.sanitize(ctx_r15.getData(row_r14, column_r16.name)), ɵngcc0.ɵɵsanitizeHtml);
} }
function NgTableComponent_tr_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵtemplate(1, NgTableComponent_tr_6_td_1_Template, 1, 1, "td", 10);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r2.columns);
} }
var NgTableComponent = (function () {
    function NgTableComponent(sanitizer) {
        this.sanitizer = sanitizer;
        // Table values
        this.rows = [];
        // Outputs (Events)
        this.tableChanged = new core_1.EventEmitter();
        this.cellClicked = new core_1.EventEmitter();
        this.showFilterRow = false;
        this._columns = [];
        this._config = {};
    }
    Object.defineProperty(NgTableComponent.prototype, "config", {
        get: function () {
            return this._config;
        },
        set: function (conf) {
            if (!conf.className) {
                conf.className = 'table-striped table-bordered';
            }
            if (conf.className instanceof Array) {
                conf.className = conf.className.join(' ');
            }
            this._config = conf;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NgTableComponent.prototype, "columns", {
        get: function () {
            return this._columns;
        },
        set: function (values) {
            var _this = this;
            values.forEach(function (value) {
                if (value.filtering) {
                    _this.showFilterRow = true;
                }
                if (value.className && value.className instanceof Array) {
                    value.className = value.className.join(' ');
                }
                var column = _this._columns.find(function (col) { return col.name === value.name; });
                if (column) {
                    Object.assign(column, value);
                }
                if (!column) {
                    _this._columns.push(value);
                }
            });
        },
        enumerable: true,
        configurable: true
    });
    NgTableComponent.prototype.sanitize = function (html) {
        return this.sanitizer.bypassSecurityTrustHtml(html);
    };
    Object.defineProperty(NgTableComponent.prototype, "configColumns", {
        get: function () {
            var sortColumns = [];
            this.columns.forEach(function (column) {
                if (column.sort) {
                    sortColumns.push(column);
                }
            });
            return { columns: sortColumns };
        },
        enumerable: true,
        configurable: true
    });
    NgTableComponent.prototype.onChangeTable = function (column) {
        this._columns.forEach(function (col) {
            if (col.name !== column.name && col.sort !== false) {
                col.sort = '';
            }
        });
        this.tableChanged.emit({ sorting: this.configColumns });
    };
    NgTableComponent.prototype.getData = function (row, propertyName) {
        return propertyName.split('.').reduce(function (prev, curr) { return prev[curr]; }, row);
    };
    NgTableComponent.prototype.cellClick = function (row, column) {
        this.cellClicked.emit({ row: row, column: column });
    };
    /** @nocollapse */
    NgTableComponent.ctorParameters = [
        { type: platform_browser_1.DomSanitizer, },
    ];
    NgTableComponent.propDecorators = {
        'rows': [{ type: core_1.Input },],
        'config': [{ type: core_1.Input },],
        'tableChanged': [{ type: core_1.Output },],
        'cellClicked': [{ type: core_1.Output },],
        'columns': [{ type: core_1.Input },],
    };
NgTableComponent.ɵfac = function NgTableComponent_Factory(t) { return new (t || NgTableComponent)(ɵngcc0.ɵɵdirectiveInject(platform_browser_1.DomSanitizer)); };
NgTableComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgTableComponent, selectors: [["ng-table"]], inputs: { rows: "rows", config: "config", columns: "columns" }, outputs: { tableChanged: "tableChanged", cellClicked: "cellClicked" }, decls: 7, vars: 4, consts: [["role", "grid", 1, "table", "dataTable", 2, "width", "100%", 3, "ngClass"], ["role", "row"], [3, "ngTableSorting", "column", "ngClass", "sortChanged", 4, "ngFor", "ngForOf"], [4, "ngIf"], [4, "ngFor", "ngForOf"], [3, "ngTableSorting", "column", "ngClass", "sortChanged"], ["class", "pull-right fa", 3, "ngClass", 4, "ngIf"], [1, "pull-right", "fa", 3, "ngClass"], ["class", "form-control", "style", "width: auto;", 3, "placeholder", "ngTableFiltering", "tableChanged", 4, "ngIf"], [1, "form-control", 2, "width", "auto", 3, "placeholder", "ngTableFiltering", "tableChanged"], [3, "innerHtml", "click", 4, "ngFor", "ngForOf"], [3, "innerHtml", "click"]], template: function NgTableComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "table", 0);
        ɵngcc0.ɵɵelementStart(1, "thead");
        ɵngcc0.ɵɵelementStart(2, "tr", 1);
        ɵngcc0.ɵɵtemplate(3, NgTableComponent_th_3_Template, 3, 5, "th", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "tbody");
        ɵngcc0.ɵɵtemplate(5, NgTableComponent_tr_5_Template, 2, 1, "tr", 3);
        ɵngcc0.ɵɵtemplate(6, NgTableComponent_tr_6_Template, 2, 1, "tr", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵpropertyInterpolate("ngClass", ctx.config.className || "");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.columns);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showFilterRow);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.rows);
    } }, directives: [ɵngcc1.NgClass, ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc2.NgTableSortingDirective, ɵngcc3.NgTableFilteringDirective], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTableComponent, [{
        type: core_1.Component,
        args: [{
                selector: 'ng-table',
                template: "\n    <table class=\"table dataTable\" ngClass=\"{{config.className || ''}}\"\n           role=\"grid\" style=\"width: 100%;\">\n      <thead>\n        <tr role=\"row\">\n          <th *ngFor=\"let column of columns\" [ngTableSorting]=\"config\" [column]=\"column\" \n              (sortChanged)=\"onChangeTable($event)\" ngClass=\"{{column.className || ''}}\">\n            {{column.title}}\n            <i *ngIf=\"config && column.sort\" class=\"pull-right fa\"\n              [ngClass]=\"{'fa-chevron-down': column.sort === 'desc', 'fa-chevron-up': column.sort === 'asc'}\"></i>\n          </th>\n        </tr>\n      </thead>\n      <tbody>\n      <tr *ngIf=\"showFilterRow\">\n        <td *ngFor=\"let column of columns\">\n          <input *ngIf=\"column.filtering\" placeholder=\"{{column.filtering.placeholder}}\"\n                 [ngTableFiltering]=\"column.filtering\"\n                 class=\"form-control\"\n                 style=\"width: auto;\"\n                 (tableChanged)=\"onChangeTable(config)\"/>\n        </td>\n      </tr>\n        <tr *ngFor=\"let row of rows\">\n          <td (click)=\"cellClick(row, column.name)\" *ngFor=\"let column of columns\" [innerHtml]=\"sanitize(getData(row, column.name))\"></td>\n        </tr>\n      </tbody>\n    </table>\n  "
            }]
    }], function () { return [{ type: platform_browser_1.DomSanitizer }]; }, { rows: [{
            type: core_1.Input
        }], tableChanged: [{
            type: core_1.Output
        }], cellClicked: [{
            type: core_1.Output
        }], config: [{
            type: core_1.Input
        }], columns: [{
            type: core_1.Input
        }] }); })();
    return NgTableComponent;
}());
exports.NgTableComponent = NgTableComponent;
