// ng2-table/__ivy_ngcc__/components/ng-table-module.js
"use strict";
var core_1 = require('@angular/core');
var common_1 = require('@angular/common');
var ng_table_component_1 = require('./table/ng-table.component');
var ng_table_filtering_directive_1 = require('./table/ng-table-filtering.directive');
var ng_table_paging_directive_1 = require('./table/ng-table-paging.directive');
var ng_table_sorting_directive_1 = require('./table/ng-table-sorting.directive');
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./table/ng-table.component');
var ɵngcc2 = require('./table/ng-table-filtering.directive');
var ɵngcc3 = require('./table/ng-table-paging.directive');
var ɵngcc4 = require('./table/ng-table-sorting.directive');
var ɵngcc5 = require('@angular/common');
var Ng2TableModule = (function () {
    function Ng2TableModule() {
    }
    /** @nocollapse */
    Ng2TableModule.ctorParameters = [];
Ng2TableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: Ng2TableModule });
Ng2TableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function Ng2TableModule_Factory(t) { return new (t || Ng2TableModule)(); }, imports: [[common_1.CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(Ng2TableModule, { declarations: [ɵngcc1.NgTableComponent, ɵngcc2.NgTableFilteringDirective, ɵngcc3.NgTablePagingDirective, ɵngcc4.NgTableSortingDirective], imports: [ɵngcc5.CommonModule], exports: [ɵngcc1.NgTableComponent, ɵngcc2.NgTableFilteringDirective, ɵngcc3.NgTablePagingDirective, ɵngcc4.NgTableSortingDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Ng2TableModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [common_1.CommonModule],
                declarations: [ng_table_component_1.NgTableComponent, ng_table_filtering_directive_1.NgTableFilteringDirective, ng_table_paging_directive_1.NgTablePagingDirective, ng_table_sorting_directive_1.NgTableSortingDirective],
                exports: [ng_table_component_1.NgTableComponent, ng_table_filtering_directive_1.NgTableFilteringDirective, ng_table_paging_directive_1.NgTablePagingDirective, ng_table_sorting_directive_1.NgTableSortingDirective]
            }]
    }], function () { return []; }, null); })();
    return Ng2TableModule;
}());
exports.Ng2TableModule = Ng2TableModule;
